flink-ml-lib/src/main/java/org/apache/flink/ml/feature/countvectorizer/CountVectorizer.java

```diff
@@ -78,7 +80,11 @@ public CountVectorizerModel fit(Table... inputs) {
         DataStream<CountVectorizerModelData> modelData =
                 DataStreamUtils.aggregate(
                         inputData,
-                        new VocabularyAggregator(getMinDF(), getMaxDF(), getVocabularySize()));
+                        new VocabularyAggregator(getMinDF(), getMaxDF(), getVocabularySize()),
+                        Types.TUPLE(
+                                Types.LONG,
+                                Types.MAP(Types.STRING, Types.TUPLE(Types.LONG, Types.LONG))),
```

Comment:
It is surprising that `TypeExtractor.getAggregateFunctionAccumulatorType` can not automatically derive the right type from `Tuple2<Long, Map<String, Tuple2<Long, Long>>>`, which only contains common types.

Can you check with related developers whether this is expected? Maybe open a ticket in Flink to track this issue.

Reply:
IMO, since we are using generic types like Map, so the internal data types are erased during the compilation, and Flink doesn't know which TypeInformation to use in the Map.

Anyway, I agree to confirm with the related developers to find out if there's a better solution.